scripts/firedrake-install

```diff
@@ -922,6 +926,32 @@ def build_and_install_libsupermesh():
         log.info("No need to rebuild libsupermesh")
 
 
+def build_and_install_tinyasm():
+    log.info("Installing TinyASM")
+    if os.path.exists("TinyASM"):
+        log.info("Updating the git repository for TinyASM")
+        with directory("TinyASM"):
+            check_call(["git", "fetch"])
+            git_sha = check_output(["git", "rev-parse", "HEAD"])
+            git_sha_new = check_output(["git", "rev-parse", "@{u}"])
+            tinyasm_changed = git_sha != git_sha_new
+            if tinyasm_changed:
+                check_call(["git", "reset", "--hard"])
+                check_call(["git", "pull"])
+    else:
+        git_clone("git+https://github.com/florianwechsung/TinyASM.git")
+        tinyasm_changed = True
+
+    if tinyasm_changed:
+        with directory("TinyASM"):
+            # Clean source directory
+            check_call(["git", "reset", "--hard"])
+            check_call(["git", "clean", "-f", "-x", "-d"])
+            run_pip_install(["-e", "."])
```

Comment:
We set `MPICC` and similar in the environment that this is run in, but perhaps not with absolute paths?

Reply:
The latter probably matters because I think we don't push the `PATH` on?